mapeditor/mainwindow.h

```diff
@@ -142,4 +144,11 @@ public slots:
 	int mapLevel = 0;
 
 	std::set<int> catalog;
+
+	// command line options
+	QString mapFilePath;			// FilePath to the H3 or VCMI map to open
+	bool extractArchives = false;	// extract original H3 archives into a separate folder
+	bool splitDefs = false;			// splits TwCrPort, CPRSMALL, FlagPort, ITPA, ITPt, Un32 and Un44 into individual PNG's
+	bool convertPcxToPng = false;	// converts single Images (found in Images folder) from .pcx to png.
+	bool deleteOriginals = false;	// delete original files, for the ones splitted / converted.
```

Comment:
I'd suggest following:
1. Make widget with checks and paths related to resources converted
2. Call this widget from in-app menu
3. Let user specify all those options and then press ok

If you still want to have this as command line options, then it's even easier. All your new code is executed inside MainWindow constructor. You should get rid of all those class members. It could be something like this:
```
struct ExtractionOptions
{
	ExtractionOptions(const QStringList & args);
	bool extractArchieves, splitDefs, convertPcxToPng, deleteOriginals;
};

MainWindow::MainWindow(...)
{
	...
	ExtractionOptions opts(qApp->arguments());
	if(opts.extractArchives)
		convertExtractedResourceFiles(splitDefs, convertPcxToPng, deleteOriginals);
}

```

Reply:
Removed from MainWindow Class